File: tools/src/main/java/com/mystery0/tools/ImageLoader/ImageCache.java
package com.mystery0.tools.ImageLoader;

import android.content.Context;
import android.graphics.Bitmap;
import android.util.LruCache;

import com.android.volley.toolbox.ImageLoader;

public class ImageCache implements ImageLoader.ImageCache
{
	private LruCache<String, Bitmap> lruCache;
	private DiskCache diskCache;
	private String fileName;

	public ImageCache(Context context)
	{
		this(context, null);
	}

	public ImageCache(Context context, String fileName)
	{
		this.fileName = fileName;
		lruCache = new LruCache<String, Bitmap>((int) (Runtime.getRuntime().maxMemory() / 1024) / 4)
		{
			@Override
			protected int sizeOf(String key, Bitmap value)
			{
				return value.getRowBytes() * value.getHeight() / 1024;
			}
		};
		diskCache = new DiskCache(context, fileName);
	}

	@Override
	public Bitmap getBitmap(String url)
	{
		Bitmap bitmap = lruCache.get((fileName == null || fileName.equals("") ? url : fileName));
		if (bitmap == null)
		{
			bitmap = diskCache.getBitmap((fileName == null || fileName.equals("") ? url : fileName));
		}
		return bitmap;
	}

	@Override
	public void putBitmap(String url, Bitmap bitmap)
	{
		lruCache.put((fileName == null || fileName.equals("") ? url : fileName), bitmap);
		diskCache.putBitmap((fileName == null || fileName.equals("") ? url : fileName), bitmap);
	}
}

File: tools/src/main/java/com/mystery0/tools/ImageLoader/ImageCache.kt
package com.mystery0.tools.ImageLoader

import android.content.Context
import android.graphics.Bitmap
import android.util.LruCache
import com.android.volley.toolbox.ImageLoader

class ImageCache internal constructor(context: Context,
									  private var fileName: String?) : ImageLoader.ImageCache
{
	private val lruCache: LruCache<String, Bitmap>
	private val diskCache: DiskCache

	init
	{
		this.fileName = fileName
		lruCache = object : LruCache<String, Bitmap>((Runtime.getRuntime().maxMemory() / 1024).toInt() / 4)
		{
			override fun sizeOf(key: String, value: Bitmap): Int
			{
				return value.rowBytes * value.height / 1024
			}
		}
		diskCache = DiskCache(context, fileName)
	}

	override fun getBitmap(url: String?): Bitmap
	{
		var bitmap = lruCache.get(if (fileName == "") url else fileName)
		if (bitmap == null)
		{
			bitmap = diskCache.getBitmap(if (fileName == "") url else fileName)
		}
		return bitmap
	}

	override fun putBitmap(url: String?, bitmap: Bitmap?)
	{
		lruCache.put(if (fileName == "") url else fileName, bitmap)
		diskCache.putBitmap(if (fileName == "") url else fileName, bitmap)
	}
}